### PR TITLE
Changes wizard checks and fix wizard apprentices

### LINKED ISSUE #2047: Polymorph messes with antags
If, for whatever reason, you want to have more antag types than just wizard, for a gimmick or something, polymorph will screw over the secondary antag. A vampire lost his vampire powers to this, but was still flagged as vampire. Couldn't even drink blood, glare or hypnotise. Yet to be confirmed for other antags like changelings but I'm betting it can happen still.

### PR DESCRIPTION
**Pull request by Alberyk**
-added a check to see if the mob is a wizard or apprentice
-replaced the faction check with the new check
-fixed wizard apprentices, also giving them a book with three spells uses
-you can now cast cure major wounds on yourself
-reduces the brain damage caused by swap
Fixes #2047

### CODE REVIEW

File: code/modules/spells/spellbook/student.dm [hunk 1]
```diff
@@ -1,23 +1,22 @@
 /obj/item/weapon/spellbook/student
 	spellbook_type = /datum/spellbook/student
 
+/obj/item/weapon/spellbook/student/attack_self(mob/living/user as mob)
+	if(user.mind.assigned_role == "Space Wizard")
```
Comment (Lohikar) [inline code/modules/spells/spellbook/student.dm]: This could probably use `is_wizard(TRUE)`.

Comment (skull132) [inline code/modules/spells/spellbook/student.dm]: What he said.

**Comment by Alberyk** [comment]
It is fixed already.

File: code/modules/spells/spellbook/student.dm [hunk 2]
```diff
@@ -1,23 +1,22 @@
...
-	Most upgrades fall into two categories, 'Speed' and 'Power'. Speed upgrades decrease the time you have to spend recharging your spell. \
-	Power increases the potency of your spells. Spells are also special in that they can be refunded while inside the Wizard Acadamy, so if you want to test a spell out before moving out into the field, feel free to do that in the comfort of our \
-	home.<br>Artifacts, or 'Artefacts' as we call them, are powerful wizard tools or items made specially for wizards everywhere. Extremely potent, they cannot be refunded like spells, and some of them can be used by non-wizards, \
-	so be careful!<br>Knowing these three concepts puts you in a league above most wizards, however knowledge of spells is just as important so we've included a list of spells below made specifically for the beginning wizard. Take all of them, or mix and match, \
-	remember being creative is half of being a wizard!"
-	book_flags = CAN_MAKE_CONTRACTS
-	max_uses = 5
+	title_desc = "Buy spells using your available spell slots. Artefacts may also be bought however their cost is permanent."
+	book_flags = null
```
Comment (skull132) [inline code/modules/spells/spellbook/student.dm]: Bitfields should be left as `0`. A wild `null` can lead to datatype incompatibility errors.

Comment (Alberyk) [inline code/modules/spells/spellbook/student.dm]: Got it.

**Pull request merged**